plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts

```diff
@@ -43,39 +48,61 @@ const permissionCriteriaSchema: z.ZodSchema<
   ]),
 );
 
-const applyConditionsRequestSchema = z.object({
-  resourceRef: z.string(),
-  resourceType: z.string(),
-  conditions: permissionCriteriaSchema,
-});
+const applyConditionsRequestSchema = z.array(
+  z.object({
+    id: z.string(),
+    resourceRef: z.string(),
+    resourceType: z.string(),
+    conditions: permissionCriteriaSchema,
+  }),
+);
 
 /**
  * A request to load the referenced resource and apply conditions in order to
  * finalize a conditional authorization response.
  *
  * @public
  */
-export type ApplyConditionsRequest = {
+export type ApplyConditionsRequestEntry = Identified<{
   resourceRef: string;
   resourceType: string;
   conditions: PermissionCriteria<PermissionCondition>;
-};
+}>;
+
+/**
+ * A batch of {@link ApplyConditionsRequestEntry} objects.
+ *
+ * @public
+ */
+export type ApplyConditionsRequest = ApplyConditionsRequestEntry[];
 
 /**
  * The result of applying the conditions, expressed as a definitive authorize
  * result of ALLOW or DENY.
  *
  * @public
  */
-export type ApplyConditionsResponse = {
-  result: AuthorizeResult.ALLOW | AuthorizeResult.DENY;
-};
+export type ApplyConditionsResponseEntry = Identified<DefinitivePolicyDecision>;
+
+/**
+ * A batch of {@link ApplyConditionsResponseEntry} objects.
+ *
+ * @public
+ */
+export type ApplyConditionsResponse = ApplyConditionsResponseEntry[];
 
 const applyConditions = <TResource>(
   criteria: PermissionCriteria<PermissionCondition>,
   resource: TResource,
   getRule: (name: string) => PermissionRule<TResource, unknown>,
 ): boolean => {
+  // If resource was not found, deny. This avoids leaking information from the
+  // apply-conditions API which would allow a user to differentiate between
+  // non-existent resources and resources to which they do not have access.
+  if (typeof resource === 'undefined') {
```

Comment:
why check `typeof resource` here?

Reply:
It's a force of habit for me since it's always safe - it can't throw or give incorrect results, whereas `resource === undefined` can do both (some interesting commentary on the details [here](https://stackoverflow.com/a/4726198)). Using `typeof` is rare in this codebase though, and it doesn't add any value given the constraints of this environment (running in a known version of node, compiled with TS). Happy to switch over to the more standard approach!

Reply:
Done in b3507f11cc.

Reply:
ah good to know! don't have strong feelings either way, was mostly curious :)